Data/HashMap/Base.hs

```diff
@@ -1209,6 +1575,19 @@ lookupInArray k0 ary0 = go k0 ary0 0 (A.length ary0)
                 | otherwise -> go k ary (i+1) n
 {-# INLINABLE lookupInArray #-}
 
+-- | /O(n)/ Lookup the value associated with the given key in this
+-- array.  Returns 'Nothing' if the key wasn't found.
+lookupInArrayWithPosition :: Eq k => k -> A.Array (Leaf k v) -> Maybe (v, Int)
```

Comment:
What is this for?

Reply:
Its used to lookup the key in the collision array (same as `lookupInArray`) but returns the value along with the collision position. It should be more clear now that I've switched to the `LookupRes a` type. It's called from the `Collision` case in `lookupRecordCollision`.